src/test/Components/CreateImageWizard/formComponents/TargetEnvironment.test.js

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
```

Comment:
Looks good and also brings up question about our testing strategy. Do we want to test components of a component individually? Do we want to test scenarios? Happy paths? Again, not really the main aim of this PR, but wanted to put this here as it might be a good idea to have a discussion about it in the not so distant future.

Reply:
Thanks for bringing that up Klara! So I have an idea for our tests that I'm starting to experiment here.

I think that Ideally, unit tests and integration tests have to coexist together. So the tests we have now are relevant and are not to be replaced, but they can't test all the paths, it's just too much. Having tests that are focusing on smaller areas means that these tests can test more without exploding in complexity. And then the intersection of these tests is what is interesting and provides more confidence. Because if you test extensively two components A and B, and you get an integration test testing A+B, you don't need to test all the combination of integrating A and B together, their behavior is tested to be robust independently, so a few happy path for A+B should be enough.

For instance in this PR I wanted to test the new Arch64 functionality. I could have added a bunch of tests in the integration testing, like doing an end to end test with arch the same way we are doing it with x86. But it would have meant duplicating tests parts that aren't specific to the change of architecture.

I decided instead to rely on the fact that the wizard is tested already with x86. So I just needed to add specific testing for Arch somewhere, and the combination of the happy path for x86 + the combinatorics tests for arch and x86 would bring me confidence that the FSC step is robust.

The question is, do you like it? What's your opinion about it?

Reply:
I see! Completely agree that we can't possibly test every happy path in the integration tests as the complexity would truly be through the roof. I'm trying to think about what could we possibly test outside of the first step in this case. Nothing comes to mind as a lot of things around architecture are tucked in the background.

I do like the idea of more comprehensive tests for smaller parts and then few happy paths to check if everything fits together well! We can probably discuss the specifics about our testing strategy even more later, but this definitely sounds like a good approach to me :+1: